server/lib/get-notify-status-view-data.js

```diff
@@ -0,0 +1,40 @@
+const fetch = require('node-fetch')
+const { getComponentTag, getServiceTag } = require('./helpers')
+
+/**
+ * Get the status summary of GOV.UK Notify.
+ *
+ * @returns {object} the data from
+ * https://status.notifications.service.gov.uk/api/v2/summary.json including an
+ * additional property `lastChecked` to indicate when the request was made.
+ */
+async function getNotifyStatus () {
+  const url = 'https://status.notifications.service.gov.uk/api/v2/summary.json'
+  const response = await fetch(url)
+  const data = await response.json()
+
+  data.lastChecked = Date.now()
+  return data
+}
+
+/**
+ * Formats the GOV.UK Notify status data for use in the `notify-status`
+ * partial.
+ */
+module.exports = async () => {
+  const data = await getNotifyStatus()
```

Comment:
Would a failure in fetching the status from Notify cause the whole 'send a message' page to error? Is it worth checking handling any HTTP errors from the status endpoint and reporting the system as unreachable, rather than let the errors bubble up?

Reply:
Yes, good point.